[EDU-3133] feat: manage bots guide (#472)
aziontech/docs: src/content/docs/en/pages/secure-journey/advanced-configurations/manage-bots.mdx

---
title: How to manage bots
description: Protect your applications and its contents from bots attacks with a Marketplace integration.
meta_tags: secure, solution, bots, protect
namespace: docs_secure_manage_bots
permalink: /documentation/products/guides/secure/manage-bots/
menu_namespace: secureMenu
---

import Button from '~/components/Button.astro'

> Securing your applications and its contents from bots attacks is an important task. By doing so in Azion Computing platform, edge functions allow a serverless solution aligned with [Edge Firewall](/en/documentation/products/edge-firewall/) security policies.

Reviewer: ```suggestion
You can protect your applications and their contents from bot attacks through Azion Edge Platform. To do so, you leverage on edge functions and allow a serverless solution aligned with [Edge Firewall](/en/documentation/products/edge-firewall/) security policies to execute this task.
```

Reviewer: a primeira frase parece mais de opinião ou abstracto.

> Check [Edge Firewall integrations](/en/documentation/products/marketplace/integrations/#edge-firewall-functions) on Marketplace to find the available integrations for bot protection.

Reviewer: talvez vale usar botão para levar para a ref?

---

## Implementation

Azion provides the **Bot Protection** integration through [Azion Marketplace](/en/documentation/products/marketplace/).

It works as an **Edge Firewall** integration, needing an edge firewall configured and the **Edge Functions** module enabled.

<Button href="/en/documentation/products/guides/bot-protection/" text="go to bot protection installation guide" variant="secondary" target="_blank"></Button>

---

## Capabilities

By installing and instantiating the **Bot Protection** integration, you're able to count on a set of possible settings and customizations:

> - Edge Firewall and Rules Engine
> - Score-based analyzis
> - Identify different bots
> - Reputation intelligence
> - SIEM integration
> - Custom Rules

Reviewer: ```suggestion
-  Take advantage of the features of Azion Edge Firewall and Rules Engine to protect your network, applications, and assets from bad bots.
- Obtain a score-based analysis to identify the type of bots, based on rules and behaviors.
- Be able to identify between good and bad bots, including bad bot signatures, malicious intent detected, scripted bots, and malicious browser behavior.
- Benefit from using *Reputation Intelligence* to establish a profile of each person who visits your site.
- Integrate with your SIEM platform to receive logs.
- Customize the function to better attend to your needs.
```

---

## Practical examples

> Once you've installed and configure the **Bot Protection** integration on Edge Firewall, you can customize a few configurations to support certain cases.

Reviewer 2: ```suggestion
Once you've installed and configured the **Bot Protection** integration on Edge Firewall, you can customize a few configurations to support certain cases.
```

    "internal_logs": "2"
}
```

By adding these variables, all the requests passed through the **Bot Protection** integration will be logged and exhibited on [Real-Time Events](/en/documentation/products/real-time-events/).

You can query the **Edge Functions** dataset to analyze and identifiy possible patterns. The `message_content` variable can provide more specific information on `Remote Address`, `User-Agent`, geolocation, WAF behaviors such as matched rules, and more. 

### Integrate with observability tools

To set up an edge function, follow the steps in [Setting up the function](/en/documentation/products/guides/bot-protection/#setting-up-the-function).

Then, on the **Arguments** tab, pass the following variable to configure information for analysis on an observability tool:

```json
{
     "siem_connection_args": {
    "host": "host.requestinspector.com",
    "uri": "/",
    "port": "443",
    "scheme": "https",
    "additional_headers": {
      "X-Fake-Auth": "an authentication header"
    }
  }
}
```

With this specific configuration, you're calibrating the Bot Protection integration to send your logs to the [Request Inspector](https://requestinspector.com/) tool.

> You can insert the `/inspector/host` string on the URL path to receive the logs and its variables.

Reviewer 2: ```suggestion
You can insert the `/inspector/host` string on the URL path to receive the logs and their variables.
```

> You can also repeat this process and replace the specification for other observability tools, such as SIEM integrations.

Reviewer 2: ```suggestion
You can also repeat this process and replace the specification with other observability tools, such as SIEM integrations.
```
